pkg/auto/notificationsemail/auto.go

```diff
@@ -0,0 +1,213 @@
+// Package notificationsemail has been made to tick a box and needs improvement.
+// provides an automation that send a CSV list of all notifications recorded by smart core
+// in the previous month. This should be run on the 1st day of the month as it fetches the notifications for
+// the previous month, not the previous ~30 days
+// Test program for meter reading automation is in 'cmd/tools/test-notificationsemail/main.go'.
+package notificationsemail
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"sort"
+	"time"
+
+	"go.uber.org/zap"
+
+	"github.com/vanti-dev/sc-bos/pkg/auto"
+	"github.com/vanti-dev/sc-bos/pkg/auto/notificationsemail/config"
+	"github.com/vanti-dev/sc-bos/pkg/gen"
+	"github.com/vanti-dev/sc-bos/pkg/task"
+	"github.com/vanti-dev/sc-bos/pkg/task/service"
+)
+
+const AutoName = "notificationsemail"
+
+var Factory auto.Factory = factory{}
+
+type factory struct{}
+
+type autoImpl struct {
+	*service.Service[config.Root]
+	auto.Services
+}
+
+func (f factory) New(services auto.Services) service.Lifecycle {
+	a := &autoImpl{Services: services}
+	a.Service = service.New(service.MonoApply(a.applyConfig), service.WithParser(config.ReadBytes))
+	a.Logger = a.Logger.Named(AutoName)
+	return a
+}
+
+// getAlertsInLastMonth gets the alerts that have happened in the previous month (not the last 30 days)
+// this automation is intended to be run on the 1st of the month
+func (a *autoImpl) getAlertsInLastMonth(ctx context.Context, alertClient gen.AlertApiClient, name string) []*gen.Alert {
+	var lastMonth []*gen.Alert
+
+	listAlerts := gen.ListAlertsRequest{
+		Name: name,
+	}
+
+	res, err := alertClient.ListAlerts(ctx, &listAlerts)
+
+	for _, a := range res.Alerts {
+		// this auto is designed to be run on the 1st dat of the month, so -1 gets all last months
+		if a.CreateTime.AsTime().Month() == (time.Now().Month() - 1) {
```

Comment:
This doesn't work on the 1st Jan: see this example https://go.dev/play/p/_emI5EqJfDq